### PR TITLE
Collapse page-layout twin hooks, widget factories and relation cards

### PR DESCRIPTION
**Pull request by FelixMalfait**
## What

Four independent collapses of duplicated code under `packages/twenty-front/src/modules/page-layout` (one commit each). Net: 58 files, +1564 / −3015.

### 1. Up/down widget move hooks (12 files, +478 / −946)

- Removed: `hooks/useMovePageLayoutWidgetUp.ts`, `hooks/useMovePageLayoutWidgetDown.ts`, `hooks/useCanMovePageLayoutWidgetUp.ts`, `hooks/useCanMovePageLayoutWidgetDown.ts` and their four test files (801 lines).
- Kept: `hooks/useMovePageLayoutWidget.ts` returning `movePageLayoutWidget(widgetId, direction)` and `hooks/useCanMovePageLayoutWidget.ts` returning `canMovePageLayoutWidget(widgetId, direction)` with `direction: 'up' | 'down'`, the literal `getAdjacentFitContentWidgetIndex` already took.
- Tests: `__tests__/useMovePageLayoutWidget.test.tsx` and `__tests__/useCanMovePageLayoutWidget.test.tsx` are `describe.each` over both directions and keep all 26 original cases (12 move, 14 can-move), including the two direction-specific edge cases around viewport-filling widgets.
- Callers updated: `side-panel/pages/page-layout/hooks/useWidgetSettingsPlacement.ts`, `side-panel/pages/page-layout/components/WidgetSettingsPlacementSection.tsx`.

### 2. Dashboard widget creation hooks and draft widget factories (23 files, +681 / −1284)

- Removed hooks: `hooks/useCreatePageLayout{Iframe,FrontComponent,RecordTable,StandaloneRichText}Widget.ts` (same ~90-line pipeline, different factory).
- Removed factories: `utils/createDefault{Iframe,FrontComponent,RecordTable,Graph}Widget.ts` (their only consumer was a removed hook) and the FrontComponent/RecordTable/Graph factory tests (Iframe had none).
- Added: `hooks/useCreatePageLayoutWidget.ts` — `createPageLayoutWidget({ type, title, configuration, objectMetadataId? })` owns the pipeline (default position from the dragged area, layout registration, `addWidgetToTab`, dragged-area reset). Sizes resolve as `WIDGET_SIZES[type]` first, then `getWidgetSize(configuration.configurationType)`, the precedence `buildTabWidgetLayouts` already uses, so every per-type default and minimum size is unchanged.
- Added: `utils/buildDraftPageLayoutWidget.ts` fills the 18-field `PageLayoutWidget` skeleton once (grid or vertical-list position); `utils/buildDefaultBarChartConfiguration.ts` carries the bar chart defaults that lived inside `createDefaultGraphWidget`.
- Kept: `hooks/useCreatePageLayoutGraphWidget.ts` as a thin adapter (175 → 92 lines) over the generic hook; `utils/createDefault{StandaloneRichText,Field,Fields}Widget.ts` become one-call wrappers over `buildDraftPageLayoutWidget` (their record-page consumers and tests are untouched).
- Tests: `utils/__tests__/buildDraftPageLayoutWidget.test.ts` (skeleton, grid/vertical-list typenames, objectMetadataId), `utils/__tests__/buildDefaultBarChartConfiguration.test.ts`, `hooks/__tests__/useCreatePageLayoutWidget.test.tsx` (active tab + isolated layouts, dragged area grown to the minimum, configuration-size fallback, layout accumulation, throw without a tab), `hooks/__tests__/useCreatePageLayoutGraphWidget.test.tsx` (bar chart config/size, title numbered after existing vertical bar charts only).
- Callers updated: `side-panel/pages/page-layout/components/SidePanelPageLayoutDashboardWidgetTypeSelect.tsx` now passes type/title/configuration inline; `widgets/record-table/hooks/__tests__/useAddDraftViewForRecordTableWidget.test.tsx` builds its fixture with `buildDraftPageLayoutWidget`.

### 3. Call-recording summary/transcript twins (12 files, +57 / −163)

- Removed: `call-recording/hooks/useCallRecordingForSummary.ts` + `useCallRecordingForTranscript.ts`, `call-recording-summary/hooks/useCallRecordingSummaryWidgetData.ts` + `call-recording-transcript/hooks/useCallRecordingTranscriptWidgetData.ts`, `call-recording-summary/components/CallRecordingSummaryWidget.tsx` + `call-recording-transcript/components/CallRecordingTranscriptWidget.tsx`.
- Kept, in the shared `widgets/call-recording/` module: `hooks/useCallRecordingForWidget.ts`, `hooks/useCallRecordingWidgetData.ts`, `components/CallRecordingWidget.tsx`, `types/CallRecordingWidgetKind.ts` (`'summary' | 'transcript'`). The record fields per kind are unchanged, and the video field is still only requested for the transcript when the role may read it.
- Callers updated: `CallRecordingSummaryWidgetContent`, `CallRecordingTranscriptWidgetContent`, `WidgetActionCallRecordingSummary`, `WidgetActionCallRecordingTranscript`, `widgets/components/WidgetContentRenderer.tsx`.

### 4. Relation card and chip triplets (11 files, +348 / −622)

- Added: `field/components/FieldWidgetRelationRecordsCard.tsx` (provider stack, read-only wiring, show-more pagination, list; takes `relationRecords` and `isReadOnly`), `field/components/FieldWidgetRelationRecordChips.tsx`, `field/hooks/useFieldWidgetJunctionRelationRecords.ts` (the junction derivation both junction adapters shared), `field/types/FieldWidgetRelationRecord.ts`, `field/types/FieldWidgetRelationCardRecord.ts`.
- Kept as adapters with the same names and props (so `FieldWidget.tsx` is untouched): `FieldWidget{Relation,MorphRelation,JunctionRelation}Card.tsx` 182/184/197 → 38/37/38 lines, `FieldWidget{Relation,MorphRelation,JunctionRelation}Field.tsx` 75/59/75 → 44/36/31 lines. The junction card is still forced read-only, through the shell's `isReadOnly` prop, with the original reason kept as a comment.

## Why

Measured duplication before this PR: the four up/down hooks (61/61 and 57/57 lines) differed only in the word `up`/`down`; the five creation hooks (700 lines) shared the same body and differed in the factory they called; the seven `createDefault*Widget` factories (363 lines) each restated the same 18-field skeleton; the call-recording twins were 40/40 (3 lines differ), 68/75 (field list) and 23/23; the three relation cards (563 lines) and three relation fields (209 lines) shared the whole shell and differed only in how the records were derived. Every fix to any of these had to land two to five times.

No product change is intended. The few internal differences are listed under "Left out / judgment calls".

## Verification

Run from the worktree, in this order, on the final tree:

- `cd packages/twenty-front && npx oxfmt src/modules/page-layout src/modules/side-panel/pages/page-layout`, then `npx oxfmt --check` on the same paths → `All matched files use the correct format.`
- `npx oxlint --type-aware -c .oxlintrc.json src/modules/page-layout src/modules/side-panel/pages/page-layout` → `Found 0 warnings and 0 errors.` (1269 files, 29 rules)
- `npx nx typecheck twenty-front --skip-nx-cache` (worktree root) → `Successfully ran target typecheck for project twenty-front and 10 tasks it depends on` (run on the final tree, after the last commit's changes)
- `cd packages/twenty-front && npx jest <files>` over every rewritten test and every test whose subject changed → 15 suites, 80 tests, all passing:
  `hooks/__tests__/useMovePageLayoutWidget.test.tsx`, `hooks/__tests__/useCanMovePageLayoutWidget.test.tsx`, `hooks/__tests__/useCreatePageLayoutWidget.test.tsx`, `hooks/__tests__/useCreatePageLayoutGraphWidget.test.tsx`, `hooks/__tests__/useCreateRecordPageFieldWidgets.test.tsx`, `hooks/__tests__/useCreateRecordPageNoteWidget.test.tsx`, `utils/__tests__/buildDraftPageLayoutWidget.test.ts`, `utils/__tests__/buildDefaultBarChartConfiguration.test.ts`, `utils/__tests__/createDefaultStandaloneRichTextWidget.test.ts`, `utils/__tests__/createDefaultFieldWidget.test.ts`, `utils/__tests__/createDefaultFieldsWidget.test.ts`, `widgets/record-table/hooks/__tests__/useAddDraftViewForRecordTableWidget.test.tsx`, `widgets/call-recording/hooks/__tests__/useCallRecordingWidgetUnavailableReason.test.tsx`, `components/__tests__/PageLayoutVerticalList.test.tsx`, `side-panel/pages/page-layout/hooks/__tests__/useWidgetSettingsPlacementSelectableItemIds.test.tsx`.
- Baseline before any change: the same lint command reported 0/0 and the 6 pre-existing affected suites (23 tests) passed, so nothing here inherits a red state.
- Not run, per scope: storybook and the full jest suite.

## Left out / judgment calls

- `useCreatePageLayoutGraphWidget` stays as a thin adapter rather than being inlined at the call site: its title numbers the existing vertical bar charts from the live draft, which the select component cannot read after `removePageLayoutWidgetAndPreservePosition` has mutated the store in the same handler.
- `createDefault{StandaloneRichText,Field,Fields}Widget` are kept as wrappers instead of inlining their configuration literals: each has two or three record-page consumers (`useCreateRecordPageNoteWidget`, `useCreateRecordPageFieldWidget`, `useCreateRecordPageFieldsWidget`, `SidePanelPageLayoutRecordPageWidgetTypeSelect`, a story) that are not part of the grid-creation pipeline.
- The record-table draft no longer sets `isOverridden: false`. The field is deprecated in the schema, nothing in the front reads it, and `convertPageLayoutDraftToUpdateInput` never sent it.
- The missing-active-tab error is now `A tab must be selected to create a new ${type} widget` (e.g. `... new IFRAME widget`) instead of five hand-written variants.
- `useCreatePageLayoutWidget` reads the active tab id from the store at call time for every type (the rich-text and graph hooks already did; the iframe, front-component and record-table hooks subscribed at render time). The record-table path now uses the same `tabListInstanceId` as its siblings instead of deriving it from the page layout id alone; the dashboard widget type select is only reachable from dashboard contexts, where both derivations yield the same id.
- Relation card list keys are `${record.id}-${fieldMetadataId}` for all three kinds (previously `record.id`, `${id}-${fieldMetadataId}` and `record.id`); uniqueness is the same in each case.
- The card shell calls `useIsRecordFieldReadOnly` for junction cards too (the junction card previously skipped it) and ORs the result with the forced `isReadOnly`, so junction cards remain read-only regardless.
- `CallRecordingWidget` lives in the shared `call-recording/` module and imports the two kind-specific content components; the alternative was keeping two 7-line wrappers.
- The dashboard rich-text widget keeps its untranslated `'Untitled Rich Text'` title, previously supplied by the factory default.
- The former factory case "objectMetadataId from fieldSelection over params" no longer exists as a concept (the graph adapter only ever passed the field selection's id); the graph hook test asserts the objectMetadataId comes from the field selection.
- No new tests for the call-recording hooks (none existed; they are Apollo-bound) or for the relation shells/adapters (none existed); tests were added only where collapsed code already had them or where per-type sizing needed locking.

<a href="https://cubic.dev/pr/twentyhq/twenty/pull/25259?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: packages/twenty-front/src/modules/page-layout/widgets/field/components/FieldWidgetJunctionRelationCard.tsx [hunk 1]
```diff
@@ -1,37 +1,8 @@
...
-import { useTargetRecord } from '@/ui/layout/contexts/useTargetRecord';
-import { SidePanelProvider } from '@/ui/layout/side-panel/contexts/SidePanelContext';
-import { isDefined } from 'twenty-shared/utils';
-import { themeCssVariables } from 'twenty-ui/theme-constants';
-
-const StyledShowMoreButtonContainer = styled.div`
-  padding-top: ${themeCssVariables.spacing[2]};
-`;
+import { FieldWidgetRelationRecordsCard } from '@/page-layout/widgets/field/components/FieldWidgetRelationRecordsCard';
+import { useFieldWidgetJunctionRelationRecords } from '@/page-layout/widgets/field/hooks/useFieldWidgetJunctionRelationRecords';
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/page-layout/widgets/field/components/FieldWidgetJunctionRelationCard.tsx:6]: Not changed here: the junction card passed `relationFieldMetadataId=""` on main too (`FieldWidgetJunctionRelationCard.tsx` line 178), so this PR preserves it verbatim. Passing the junction field's id changes what the expanded target excludes and belongs in its own PR.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: packages/twenty-front/src/modules/page-layout/widgets/field/components/FieldWidgetJunctionRelationCard.tsx [hunk 2]
```diff
@@ -46,152 +17,24 @@ export const FieldWidgetJunctionRelationCard = ({
...
-          </FieldInputEventContext.Provider>
-        </FieldContext.Provider>
-      </RecordFieldsScopeContextProvider>
-    </SidePanelProvider>
+    <FieldWidgetRelationRecordsCard
+      fieldDefinition={fieldDefinition}
+      relationRecords={junctionRelationRecords.map(
+        (junctionRelationRecord) => ({
+          ...junctionRelationRecord,
+          fieldMetadataId: '',
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/page-layout/widgets/field/components/FieldWidgetJunctionRelationCard.tsx:33]: Not changed here: `""` is what the junction card passed before the shell was shared (main, line 178). Wiring the real field id is a behaviour change for a separate PR.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: packages/twenty-front/src/modules/page-layout/widgets/field/components/FieldWidgetRelationRecordChips.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+export const FieldWidgetRelationRecordChips = ({
+  relationRecords,
+  isInSidePanel,
+}: FieldWidgetRelationRecordChipsProps) => (
+  <SidePanelProvider value={{ isInSidePanel }}>
+    <StyledContainer>
+      <StyledRelationChipsContainer>
+        {relationRecords.map(({ record, objectNameSingular }) => (
+          <RecordChip
+            key={record.id}
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/page-layout/widgets/field/components/FieldWidgetRelationRecordChips.tsx]: Done in 791327562c: chip keys are `objectNameSingular-record.id`.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: packages/twenty-front/src/modules/page-layout/widgets/field/components/FieldWidgetRelationRecordsCard.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+            useUpdateRecord: useUpdateOneObjectRecordMutation,
+            isRecordFieldReadOnly: isReadOnly || isRecordFieldReadOnly,
+          }}
+        >
+          <FieldInputEventContext.Provider value={{ onSubmit: handleSubmit }}>
+            <StyledWidgetContentContainer>
+              <RecordDetailRecordsListContainer>
+                {visibleRelationRecords.map(
+                  ({ record, objectNameSingular, fieldMetadataId }) => (
+                    <Fragment key={`${record.id}-${fieldMetadataId}`}>
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/page-layout/widgets/field/components/FieldWidgetRelationRecordsCard.tsx]: Done in 791327562c: card keys now include the object name as well.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: packages/twenty-front/src/modules/page-layout/hooks/__tests__/useMovePageLayoutWidget.test.tsx [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+          result.current.movePageLayoutWidget(movedWidgetId, direction);
+        });
+
+        const draft = store.get(getDraftAtom());
+
+        expect(draft.tabs[0].widgets.map((widget) => widget.id)).toEqual([
+          'widget-b',
+          'widget-a',
+          'timeline-widget',
+        ]);
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/page-layout/hooks/__tests__/useMovePageLayoutWidget.test.tsx:133]: The assertion is on the canonicalised list, where the viewport-filling timeline is placed last for both directions; the case passes locally and in CI for `down` too.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: packages/twenty-front/src/modules/page-layout/widgets/call-recording/hooks/useCallRecordingForWidget.ts [hunk 6]
```diff
@@ -41,6 +43,15 @@ export const useCallRecordingForTranscript = (): {
     refetchCallRecordingId,
   } = useCallRecordingIdForWidget({ skip: shouldSkipQuery });
 
+  const recordFields = CALL_RECORDING_RECORD_FIELDS_BY_WIDGET_KIND[kind];
+
+  // The transcript widget plays the video, but requesting a field the role
+  // cannot read fails the whole query.
+  const recordGqlFields =
+    kind === 'transcript' && !isFieldRestricted('video')
+      ? { ...recordFields, video: true }
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/page-layout/widgets/call-recording/hooks/useCallRecordingForWidget.ts]: Done in 791327562c: the transcript-with-video field map is a module-level constant, so the hook selects between two stable maps as the original did.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: packages/twenty-front/src/modules/page-layout/widgets/field/components/FieldWidgetRelationRecordChips.tsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  relationRecords: FieldWidgetRelationRecord[];
+  isInSidePanel: boolean;
+};
+
+export const FieldWidgetRelationRecordChips = ({
+  relationRecords,
+  isInSidePanel,
+}: FieldWidgetRelationRecordChipsProps) => (
+  <SidePanelProvider value={{ isInSidePanel }}>
+    <StyledContainer>
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/page-layout/widgets/field/components/FieldWidgetRelationRecordChips.tsx]: Done in 791327562c: the chips return nothing for an empty relation, as the original relation and morph fields did.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: packages/twenty-front/src/modules/page-layout/widgets/field/components/FieldWidgetRelationRecordChips.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/page-layout/widgets/field/components/FieldWidgetRelationRecordChips.tsx]: Correction: the change is in 664c45e0e4; the commit named above was the previous head, the push had not gone through yet.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: packages/twenty-front/src/modules/page-layout/widgets/field/components/FieldWidgetRelationRecordsCard.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/page-layout/widgets/field/components/FieldWidgetRelationRecordsCard.tsx]: Correction: the change is in 664c45e0e4; the commit named above was the previous head, the push had not gone through yet.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: packages/twenty-front/src/modules/page-layout/widgets/call-recording/hooks/useCallRecordingForWidget.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/page-layout/widgets/call-recording/hooks/useCallRecordingForWidget.ts]: Correction: the change is in 664c45e0e4; the commit named above was the previous head, the push had not gone through yet.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: packages/twenty-front/src/modules/page-layout/widgets/field/components/FieldWidgetRelationRecordChips.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/page-layout/widgets/field/components/FieldWidgetRelationRecordChips.tsx]: Correction: the change is in 664c45e0e4; the commit named above was the previous head, the push had not gone through yet.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: packages/twenty-front/src/modules/page-layout/widgets/field/components/FieldWidgetRelationRecordsCard.tsx [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+  const useUpdateOneObjectRecordMutation = () => {
+    const updateEntity = ({
+      variables,
+    }: {
+      variables: {
+        where: { id: string };
+        updateOneRecordInput: Record<string, unknown>;
+      };
+    }) => {
+      updateOneRecord({
```
[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/page-layout/widgets/field/components/FieldWidgetRelationRecordsCard.tsx:91]: Not changed: this is moved verbatim from the three cards it replaces, each of which defined `useUpdateOneObjectRecordMutation` inline and passed it into the context the same way (`FieldWidgetRelationCard.tsx` line 80, `FieldWidgetMorphRelationCard.tsx` line 78, `FieldWidgetJunctionRelationCard.tsx` line 81 on main). Memoising it is a separate change and would apply to the other inline definitions of that pattern as well.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
